test/Oscoin/Test/HTTP/Helpers.hs

```diff
@@ -191,38 +201,48 @@ emptyBlockchain :: IsCrypto c => Blockchain c (Tx c) DummySeal
 emptyBlockchain = fromGenesis $ sealBlock "" (emptyGenesisBlock epoch defaultBeneficiary)
 
 -- | Run a 'Session' with the given initial node state
-runSession :: IsCrypto c => NodeState c -> Session c () -> Assertion
+runSession :: IsCrypto c => NodeState c DummySeal -> Session c DummySeal () -> Assertion
 runSession nst (Session sess) =
-    withNode nst $ \nh -> do
+    withNode mempty nst $ \nh -> do
         app <- API.app nh
         Wai.runSession (runReaderT sess nh) app
 
 -- | Run a 'Session' so that the blockchain state has the given
 -- Radicle bindings.
-runSessionWithState :: (IsCrypto c, MonadIO m) => [(Text, DummyPayload)] -> Session c () -> m ()
-runSessionWithState bindings (Session sess)= do
+runSessionWithState
+    :: ( IsCrypto c
+       , MonadIO m
+       , Serialise s
+       , Ord s
+       )
+    => s
+    -- ^ A seal for the genesis block.
+    -> [(Text, DummyPayload)]
+    -> Session c s ()
+    -> m ()
+runSessionWithState seal bindings (Session sess)= do
```

Comment:
What do you think about splitting this up into a polymorphic `runSessionWithState'` and a monomorphic `runSessionWithState`. This would make the function more consistent with `runSession` and `runEmptySession`. It would also eliminate the `""` value in the tests that is totally non-obvious.

Reply:
Thought about that, but was wondering if the fact this is used in only 2 places outside this module warranted the need for an extra function, but if that's your preference, I am happy to do that 😉

Reply:
6b459103